test/e2e/util/resource_queue.go

```diff
@@ -0,0 +1,100 @@
+package util
+
+import (
+	"fmt"
+	k8scontrollerclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sync"
+)
+
+type ResourceQueue struct {
+	queue       []k8scontrollerclient.Object
```

Comment:
So you might try the new standard library `containers/heap` package out: https://pkg.go.dev/container/heap#example-package-PriorityQueue